client/rpcproxy/rpcproxy.go

```diff
@@ -129,137 +127,194 @@ type RPCProxy struct {
 	// connection pool.  Pinger is an interface that wraps
 	// client.ConnPool.
 	connPoolPinger Pinger
+
+	// setServer is how new primary servers are sent to the notifier goroutine
+	setServer chan<- *ServerEndpoint
+
+	// getServer is how the primary server is sent to callers
+	getServer <-chan *ServerEndpoint
 }
 
 // NewRPCProxy is the only way to safely create a new RPCProxy.
 func NewRPCProxy(logger *log.Logger, shutdownCh chan struct{}, configInfo NomadConfigInfo, connPoolPinger Pinger) *RPCProxy {
+	setServer := make(chan *ServerEndpoint)
+	getServer := make(chan *ServerEndpoint)
 	p := &RPCProxy{
+		activatedList:  newServerList(),
+		backupServers:  newServerList(),
+		primaryServers: newServerList(),
 		logger:         logger,
 		configInfo:     configInfo,     // can't pass *nomad.Client: import cycle
 		connPoolPinger: connPoolPinger, // can't pass *nomad.ConnPool: import cycle
 		rebalanceTimer: time.NewTimer(clientRPCMinReuseDuration),
+		setServer:      setServer,
+		getServer:      getServer,
 		shutdownCh:     shutdownCh,
 	}
 
-	l := serverList{}
-	l.L = make([]*ServerEndpoint, 0)
-	p.saveServerList(l)
+	go notifier(setServer, getServer, p.shutdownCh)
 	return p
 }
 
+// notifier notifies RPC callers of the current active RPC server if there is
+// one. If not it blocks until one becomes available.
+func notifier(in <-chan *ServerEndpoint, out chan<- *ServerEndpoint, exit <-chan struct{}) {
```

Comment:
We don't need this since the `client.RPC` method is retried on error by callers.